src/printer.js

```diff
@@ -553,11 +553,12 @@ function printExpression(path, options, print) {
           node.shortForm ? "[" : "array(",
           indent(
             concat([
-              softline,
+              node.comments && node.comments.length > 0 ? hardline : softline,
```

Comment:
do we actually want to do this? seems like as long as its not more than the line length we shouldn't be enforcing new lines just because there are comments

Reply:
https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAE8OYLyYDaAjADSYD0AVJgJZQA29cmkAtmwllRZgEwBdEKRAQADjFrpkoAIYAneRADuABQUI0yELIBuEWgBNhOjMgBmshmjgiARvNlgA1nBgBlMU-oBzZDHkAV1sQQwgwCysbEXobeRhVRx82WUjrEIArNAAPACFHFzd3WU4AGWY06NFAmDEavkqQr3k47TtZOwBPBmgTMXl6GAB1IxgAC2QADgAGEX6IGyHHMW1+uDjdOBN5OABHQNodxNlk1KRLdJEbNlp-IJC0XwY4AEVAiHhGkRgOkcNx5B8b6OWhMKA+ADCEA4ZxAUGgWxEgRsABUOlpzlE4ABfbFAA

Reply:
@mgrip looks we should break only we have leading or dangling comments :confused: But improve output comment i think we should do in other PR. First goal is print all comments without break code.

Reply:
Also related https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAE8OYLyYDaA9MZgIwA6UmFANNbQEwM2YDM1AugNwh0gIABxgBLdMlABDAE4yIAdwAKshGmQgpANwiiAJv00ZkAMykAbNHAEAjGVLABrODADKQh6KgBzZDBkArtYgehBgphZWAl5WMjBK9t4AtlIRlsEAVmgAHgBC9k4urlJJcAAyXnBpUYIBMEJ1TNXBHjKxGjZSNgCe5tCGQjJeMADq+jAAFsgAHAAMAoMQViP2QhqDcLFaVQIycACOAaJ7CVLJqUhm6QJWSaJ+gcFoXt7mcACKARDwzQIwXWM9JNkCwQP4pKJzC8AMIQJIpDRQaA7EABKwAFS66kukTgAF88UA

Reply:
i agree, im just questioning if this line should change at all or just remain as `softline` always

Reply:
:+1:

Reply:
@mgrip softline break when code
```php
$array=[ // test
 1
];
```
Output:
```php
$array=[// test 1 ];
```

Reply:
ahh ok got it. ok ya agree lets just merge this to get code working and circle back on the formatting if we need to